### PR TITLE
Fix suggestions for empty indices

### LINKED ISSUE #42473: Empty index with completion suggesters returns null suggestions
**Elasticsearch version**: 6.4.2

**Plugins installed**: []

**JVM version**: java 11.0.1 2018-10-16 LTS

**OS version**: Darwin Kernel Version 18.2.0

When searching completion suggestions on a new index with no documents added, a different response is returned from when searching on na index with no documents matching.

Also a side note is that the example `PUT music` call in the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters-completion.html) does not work.

**Steps to reproduce**:

```
PUT music
{
    "mappings": {
      "_doc": {
        "properties" : {
            "suggest" : {
                "type" : "completion"
            },
            "title" : {
                "type": "keyword"
            }
        }
      }
        
    }
}
```

```
POST music/_search?pretty
{
    "suggest": {
        "song-suggest" : {
            "prefix" : "v", 
            "completion" : { 
                "field" : "suggest" 
            }
        }
    }
}
```

Returns:

```
{
  "took": 0,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 0,
    "max_score": 0,
    "hits": []
  }
}
```

```
PUT music/_doc/1?refresh
{
    "suggest" : [
        {
            "input": "Nevermind",
            "weight" : 10
        },
        {
            "input": "Nirvana",
            "weight" : 3
        }
    ]
}
```

```
POST music/_search?pretty
{
    "suggest": {
        "song-suggest" : {
            "prefix" : "v", 
            "completion" : { 
                "field" : "suggest" 
            }
        }
    }
}
```

Returns 

```
{
  "took": 0,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 0,
    "max_score": 0,
    "hits": []
  },
  "suggest": {
    "song-suggest": [
      {
        "text": "v",
        "offset": 0,
        "length": 1,
        "options": []
      }
    ]
  }
}
```

This is a problem since `SearchResponse.getSuggest()` in the JDK will return `null` in the first case and the proper `Suggest` object in the second case. I would expect it to always return `Suggest` object in both cases.

### PR DESCRIPTION
**Pull request by cbuescher**
Currently suggesters return `null` values on empty shards. Usually this gets replaced
by results from other non-epmty shards, but if the index is completely epmty (e.g. after
creation) the search responses "suggest" is also "null" and we don't render a corresponding
output in the REST response. This is an irritating edge case that requires special handling on
the user side (see #42473) and should be fixed.

This change makes sure every suggester type (completion, terms, phrase) returns at least an 
empty skeleton suggestion output, even for empty shards. This way, even if we don't find
any suggestions anywhere, we still return and output the empty suggestion.

Closes #42473

### CODE REVIEW

File: plugins/examples/custom-suggester/src/main/java/org/elasticsearch/example/customsuggester/CustomSuggester.java [hunk 1]
```diff
@@ -35,15 +36,12 @@
...
-
-        // Get the suggestion context
-        String text = suggestion.getText().utf8ToString();
+        CharsRefBuilder spare) throws IOException {
 
         // create two suggestions with 12 and 123 appended
-        CustomSuggestion response = new CustomSuggestion(name, suggestion.getSize(), "suggestion-dummy-value");
-
-        CustomSuggestion.Entry entry = new CustomSuggestion.Entry(new Text(text), 0, text.length(), "entry-dummy-value");
+        CustomSuggestion response = emptySuggstion(name, suggestion, spare);
```
Comment (javanna) [inline plugins/examples/custom-suggester/src/main/java/org/elasticsearch/example/customsuggester/CustomSuggester.java]: fox typo :)

Comment (cbuescher) [reply plugins/examples/custom-suggester/src/main/java/org/elasticsearch/example/customsuggester/CustomSuggester.java]: It wasn't me, what did I do??? :)
Cannot find the error unfortunately...

Comment (javanna) [reply plugins/examples/custom-suggester/src/main/java/org/elasticsearch/example/customsuggester/CustomSuggester.java]: emptySuggstion missing e, should be emptySuggestion ?

Comment (cbuescher) [reply plugins/examples/custom-suggester/src/main/java/org/elasticsearch/example/customsuggester/CustomSuggester.java]: Thanks, I was 🙈

**Comment by cbuescher** [comment]
@javanna thanks for the review, fixed the typo

**Review by javanna — APPROVED** [review]
LGTM

**Comment by cbuescher** [comment]
@javanna thanks for the review

**Pull request merged**